src/Control/Director.php

```diff
@@ -405,18 +406,20 @@ public static function isManifestFlushed()
     /**
      * Return the {@link SiteTree} object that is currently being viewed. If there is no SiteTree
      * object to return, then this will return the current controller.
+     * return SiteTree SilverStripe\CMS\Model\SiteTree or Controller
+     * SiteTree can't be added to docblock here as Framework does not require silverstripe/cms)
      *
-     * @return SiteTree|Controller
+     * @return mixed
      */
     public static function get_current_page()
     {
-        return self::$current_page ? self::$current_page : Controller::curr();
+        return self::$current_page ?: Controller::curr();
     }
 
     /**
      * Set the currently active {@link SiteTree} object that is being used to respond to the request.
-     *
-     * @param SiteTree $page
+     * set as SiteTree (can't be added to docblock here as Framework does not require silverstripe/cms)
```

Comment:
I'd rather put it here anyway even though PHPStan would complain to be honest. People use the CMS with framework most of the time, and ideally we remove the coupling instead.

Reply:
Personally, I would rather remove the `@param` and have a  thumbs up response from PHPStan

Reply:
It could potentially be typed as DataObject instead, and the above as DataObject|Controller. That is a semantic that is somewhat meaningful uncoupled from cms.